app/models/image.rb

```diff
@@ -25,7 +42,8 @@ def offer
   end
 
   def reset_favourite
-    favourite && imageable.images.where.not(id: id).update_all(favourite: false)
```

Comment:
How come the previous version didn't work ? 🤔

Reply:
Actually `update_all` does not trigger any callbacks, and hence it was not live updating `favourite` value of other images. Hence updated it.